client/js/test/table-view.test.js

```diff
@@ -0,0 +1,146 @@
+                     const trs = document.querySelectorAll('TBODY TR');
+                     trs[1].cells[2].click();
+
+                     // inspect the resulting state
+                     expect(formulaBarEl.value).toBe('123');
+                 });
+             });
+
+             describe('table body', () => {
+                 it('highlights the current cell when clicked', () => {
+                     // set up the initial state
+                     const model = new TableModel(10, 5);
+                     const view = new TableView(model);
+                     view.init();
+
+                     // inspect the initial state
+                     let trs = document.querySelectorAll('TBODY TR');
+                     let td = trs[2].cells[3];
+                     expect(td.className).toBe('');
+
+                     // simulate user action
+                     td.click();
+
+                     // inspect the resulting state
+                     trs = document.querySelectorAll('TBODY TR');
+                     td = trs[2].cells[3];
+                     expect(td.className).not.toBe('');
+                 });
+
+                 it('has the right size', () => {
+                     // set up initial state
+                     const numCols = 6;
+                     const numRows = 10;
+                     const model = new TableModel(numCols, numRows);
+                     const view = new TableView(model);
+                     view.init();
+
+                     // inspect the initial state
+                     let ths = document.querySelectorAll('THEAD TH');
+                     expect(ths.length).toBe(numCols);
+                 });
+
+                 it('fills in values from the model', () => {
+                     // set up the initial state
+                     const model = new TableModel(3, 3);
+                     const view = new TableView(model);
+                     model.setValue({ col: 2, row: 1 }, '123');
+                     view.init();
+                 });
+             });
+
+             describe('table header', () => {
+                 it('has valid column header labels', () => {
+                     // set up the initial state
+                     const numCols = 6;
+                     const numRows = 10;
+                     const model = new TableModel(numCols, numRows);
+                     const view = new TableView(model);
+                     view.init();
+
+                     // inspect the initial state
+                     let ths = document.querySelectorAll('THEAD TH');
+                     expect(ths.length).toBe(numCols);
+
+                     let labelTexts = Array.from(ths).map(el => el.textContent);
+                     expect(labelTexts).toEqual(['A', 'B', 'C', 'D', 'E', 'F']);
+
+                 });
+             });
+
+             describe('table footer', () => {
+                     it('sums of footer row of postive numbers', () => {
```

Comment:
Nit picking: test assertions should read as full sentences. "it sums of footer row of positive numbers" is difficult to read and know exactly what is supposed to happen. Please make sure the assertions would tell a dev unfamiliar with the project exactly what is going on in this test.